File: moviedb/src/main/java/com/ericjohnson/moviecatalogue/utils/DialogUtils.java
package com.ericjohnson.moviecatalogue.utils;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.graphics.drawable.InsetDrawable;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.WindowManager;
import android.widget.Button;
import android.widget.TextView;

import androidx.appcompat.app.AlertDialog;

import com.ericjohnson.moviecatalogue.R;

public class DialogUtils {

    public static void showCancelableDialogWithoutIcon(
            Context context,
            String title,
            String message,
            String positive,
            String negative,
            OnButtonNegativeListener listenerNegative,
            OnButtonPositiveListener listenerPositive) {
        View view = LayoutInflater.from(context).inflate(R.layout.layout_alert_dialog_without_icon, null);

        TextView tvTitle = view.findViewById(R.id.tvTitle);
        TextView tvMessage = view.findViewById(R.id.tvMessage);
        Button btnNegative = view.findViewById(R.id.btnNegative);
        Button btnPositive = view.findViewById(R.id.btnPositive);

        tvTitle.setText(title);

        if (message != null) {
            tvMessage.setVisibility(View.VISIBLE);
            tvMessage.setText(message);
        }

        if (positive != null) {
            btnPositive.setVisibility(View.VISIBLE);
            btnPositive.setText(positive);
        }

        if (negative != null) {
            btnNegative.setVisibility(View.VISIBLE);
            btnNegative.setText(negative);
        }

        AlertDialog.Builder builderWithoutIcon = new AlertDialog.Builder(context);
        builderWithoutIcon.setView(view);
        AlertDialog dialogWithoutIcon = builderWithoutIcon.create();

        if (dialogWithoutIcon.getWindow() != null) {
            dialogWithoutIcon.getWindow().setLayout(
                    WindowManager.LayoutParams.WRAP_CONTENT,
                    WindowManager.LayoutParams.WRAP_CONTENT);
            dialogWithoutIcon.getWindow().setGravity(Gravity.CENTER);
            InsetDrawable inset = new InsetDrawable(new ColorDrawable(Color.TRANSPARENT), 60);
            dialogWithoutIcon.getWindow().setBackgroundDrawable(inset);
        }

        dialogWithoutIcon.setCancelable(true);
        dialogWithoutIcon.show();

        btnPositive.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listenerPositive.onButtonPositiveClicked();
                dialogWithoutIcon.dismiss();
            }
        });

        btnNegative.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                listenerNegative.onButtonNegativeClicked();
                dialogWithoutIcon.dismiss();
            }
        });
    }

    public interface OnButtonNegativeListener {
        void onButtonNegativeClicked();
    }

    public interface OnButtonPositiveListener {
        void onButtonPositiveClicked();
    }
}

File: humanid-ui/src/main/java/com/nbs/humanidui/util/DialogUtils.kt
package com.human.android.util

import android.content.Context
import android.graphics.Color
import android.graphics.drawable.ColorDrawable
import android.graphics.drawable.InsetDrawable
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.WindowManager
import androidx.annotation.NonNull
import androidx.appcompat.app.AlertDialog
import androidx.core.content.res.ResourcesCompat
import com.human.android.util.extensions.setImageResource
import com.human.android.util.extensions.toHtml
import com.nbs.humanidui.R
import com.nbs.nucleo.utils.extensions.onClick
import com.nbs.nucleo.utils.extensions.visible
import kotlinx.android.synthetic.main.layout_alert_dialog.view.*

fun showUncancelableDialog(
        context: Context,
        icon: Int? = null,
        title: String? = null,
        message: String? = null,
        isMessageLight: Boolean = false,
        email: String? = null,
        subMessage: String? = null,
        first: String? = null, @NonNull firstListener: (() -> Unit)? = null,
        second: String? = null, @NonNull secondListener: (() -> Unit)? = null
) {
    val dialogView: View = LayoutInflater.from(context).inflate(R.layout.layout_alert_dialog, null)

    val builder = AlertDialog.Builder(context).setView(dialogView)
    val dialog = builder.create()



    with(dialogView) {
        if (icon != null) {
            imgIcon.visible()
            imgIcon.setImageResource(context, icon)
        }

        if (title != null) {
            tvTitle.visible()
            tvTitle.text = title
        }

        if (message != null) {
            tvMessage.visible()
            tvMessage.text = toHtml(message)

            if (isMessageLight) {
                tvMessage.typeface = ResourcesCompat.getFont(context, R.font.roboto_light)
            }
        }

        if (email != null) {
            tvEmail.visible()
            tvEmail.text = email
        }

        if (subMessage != null) {
            tvSubMessage.visible()
            tvSubMessage.text = subMessage
        }

        if (first != null) {
            btnFirst.visible()
            btnFirst.text = first
        }

        if (second != null) {
            btnSecond.visible()
            btnSecond.text = second
        }

        btnFirst.onClick {
            dialog.dismiss()
            firstListener?.invoke()
        }

        btnSecond.onClick {
            dialog.dismiss()
            secondListener?.invoke()
        }
    }

    dialog.apply {
        window?.apply {
            setLayout(
                    WindowManager.LayoutParams.WRAP_CONTENT,
                    WindowManager.LayoutParams.WRAP_CONTENT
            )
            setGravity(Gravity.CENTER)
            val inset = InsetDrawable(ColorDrawable(Color.TRANSPARENT), 60)
            setBackgroundDrawable(inset)
        }
        setCancelable(false)
        show()
    }
}